src/content/en/2021/caching.md

```diff
@@ -17,4 +19,141 @@ featured_stat_3: TODO
 featured_stat_label_3: TODO
 ---
 
-## TODO
+## Introduction
+
+Over the past decade ...
+
+
+## CDN Cache adoption
+
+Talk about CDN adoptions, 14% of desktop pages use a Cloudflare CDN.
+
+{{ figure_markup(
+  image="top-cdns.png",
+  caption="Adoption of the top CDNs.",
+  description="TODO",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=2035524639&format=interactive",
+  sheets_gid="58739923",
+  sql_file="top_cdns.sql"
+) }}
+
+## Service Worker adoption
+
+Talk about Service Worker adoptions, 1.2% of desktop pages registered a Service Worker, but 9% of mobile pages in the top 1000 registered a SW.
```

Comment:
<p>Adoption of Service Workers has seen a significant increase from 2020. While 1.2% of Desktop pages registered a service worker, 9% of Mobile pages in the top 1,000 registered a Service Worker.</p>